mrmustard/lab_dev/circuit_components.py

```diff
@@ -470,11 +530,37 @@ def __matmul__(self, other: CircuitComponent | Scalar) -> CircuitComponent:
         Contracts ``self`` and ``other`` without adding adjoints.
         It allows for a more custom way of contracting components.
```

Comment:
```suggestion
        Adds ``self`` and ``other``. If the representation is ``Fock`` it can add the arrays directly.
        Otherwise it will stack along the batch dimension in the ``Ansatz``.
```